pkg/controller/job/jobcontroller.go

```diff
@@ -306,6 +302,32 @@ func (jm *JobController) deletePod(obj interface{}) {
 	jm.enqueueController(job)
 }
 
+func (jm *JobController) updateJob(old, cur interface{}) {
+	oldJob := cur.(*batch.Job)
+	curJob := cur.(*batch.Job)
+
+	// never return error
+	key, _ := controller.KeyFunc(curJob)
```

Comment:
There's no point in re-trying if there was an error. Check error and just return if that happens.